### DISCUSSION TITLE
Error: No wagmi config found. Ensure you have set up a config

### DISCUSSION THREAD

**Discussion by yeshealer** [WalletConnect/walletconnect-monorepo#3562] [Developer Support] [answered]
I'm current building the dApp with latest Next.js and TypeScript. And using Tailwind CSS.

Next.js version is 13.4.19
TypeScript version is 5.1.6
Node.js version is 18.17.1
npm version is 9.6.7
yarn version is 1.22.19

I'm using the windows 11 and using the metamask for test.

I tried to using the WalletConnect 2.0 for the wallet connect module. I installed @web3modal/ethereum, @web3modal/react, wagmi, viem on latest versions.

@web3modal/ethereum is 2.7.1
@web3modal/react is 2.7.1
wagmi is 1.3.10
viem is 1.7.0

What I gonna ask the wagmi functions are broken so I can't use the them except some basic functions except getNetwork, useAccount. If I try to using the readContract or writeContract using wagmi it's return the error such as below screenshot.

![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/88679102/f1acf0b1-122c-4d39-9af6-bff0fedc8fb8)

Here is my code.
```
import { readContract } from '@wagmi/core';

export const fetchShezmuBalance = async (address: string) => {
    if (shezmuContract && address) {
        const data = await readContract({
            address: shezmuContract as `0x${string}`,
            abi: ShezmuABI,
            functionName: 'balanceOf',
            args: [address],
        })
        return Number(data) / 1e18
    }
    return 0
}
```

So I used viem readContract and writeContract functions instead of that so issue fixed but when I try to build the next.js project they are return the another error named `Error: No wagmi config found. Ensure you have set up a config`

![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/88679102/d3ba3143-b30c-4e99-8ec1-1aab8a9c93e5)

And here is the source code for Wagmi Provider set up.
```
"use client"
import { Web3Modal } from '@web3modal/react'
import { EthereumClient, w3mConnectors, w3mProvider } from '@web3modal/ethereum'
import { configureChains, createConfig, WagmiConfig } from 'wagmi'
import { arbitrum, mainnet, polygon } from 'wagmi/chains'

const chains = [arbitrum, mainnet, polygon];
const projectId = process.env.PROJECT_ID ?? (() => { throw new Error("PROJECT_ID is not defined"); })();

const { publicClient, webSocketPublicClient } = configureChains(chains, [w3mProvider({ projectId })])
const wagmiConfig = createConfig({
    autoConnect: true,
    connectors: w3mConnectors({ projectId, chains }),
    publicClient,
    webSocketPublicClient
})
const ethereumClient = new EthereumClient(wagmiConfig, chains)

interface WagmiProviderProps {
    children: React.ReactNode
}

export default function WagmiProvider({
    children
}: WagmiProviderProps) {
    return (
        <>
            <WagmiConfig config={wagmiConfig}>
                {children}
                <Web3Modal projectId={projectId} ethereumClient={ethereumClient} themeMode='dark' themeVariables={{
                    '--w3m-background-color': '#b45309',
                    '--w3m-z-index': '50'
                }} />
            </WagmiConfig>
        </>
    )
}
```

Worked on localhost successfully but can't build the project so I can't upload the data to hosting site such as Vercel or Netlify.

**Comment by glitch-txs** [comment]
Hi, could you try to update all dependencies including Viem and Wagmi to their latest versions?
If this doesn't fix the issue could you share a [MRE](https://github.com/orgs/WalletConnect/discussions/3457)? :)
Thanks!

**Comment by 0xevm1** [comment]
did you ever figure this out?

**Reply by glitch-txs** [reply, marked as answer]
delete lock file and node_modules, update wagmi and viem to the latest versions
